app/models/task.rb

```
@@ -39,14 +40,25 @@ def start_text
   end
 
   def assign!(user)
-    return AlreadyAssignedError if self.user
+    fail(AlreadyAssignedError) if self.user
 
-    update_attributes!(
+    update!(
       user: user,
       assigned_at: Time.now.utc
     )
   end
 
+  def start!
+    fail(NotAssignedError) unless assigned?
+    return if started?
```

@shanear I was torn whether this should error or just silently fail. I generally prefer "action" methods to fail if they don't mutate the data in the way they are expected. The upside here being we can blindly call `task.start!` from the show controller action. If we remove the safe check here, we'd have to add more logic to the controller